src/main/java/com/salesforce/mirus/HerderStatusMonitor.java

```diff
@@ -91,42 +90,50 @@ private void onConnectors(Throwable error, Collection<String> connectorNames) {
 
   private void processConnector(String connectorName) {
     ConnectorStateInfo stateInfo = herder.connectorStatus(connectorName);
-    if (workerId.equals(stateInfo.connector().workerId())) {
-      if (stateInfo.connector().state().equals(ConnectorStatus.State.FAILED.toString())) {
-        connectorJmxReport.incrementConnectorRestartAttempts(connectorName);
-        if (autoRestartConnectorEnabled) {
+
+    if (autoRestartConnectorEnabled) {
+      if (workerId.equals(stateInfo.connector().workerId())) {
+        if (stateInfo.connector().state().equals(ConnectorStatus.State.FAILED.toString())) {
```

Comment:
Could you add a comment about what's happening here? That only the assigned worker should restart the connector, but all workers need to process their own tasks regardless of the connector?